services/registration/src/routes/statistics.ts

```diff
@@ -2,98 +2,116 @@
 import { asyncHandler } from "@api/common";
 import express from "express";
 
-import { ApplicationModel } from "../models/application";
+import { ApplicationModel, StatusType } from "../models/application";
 import { BranchModel, BranchType } from "../models/branch";
 
 export const statisticsRouter = express.Router();
 
 statisticsRouter.route("/").get(
   asyncHandler(async (req, res) => {
-    let totalUsers = 0;
-    let confirmedUsers = 0;
-    const submittedUsers = await ApplicationModel.countDocuments({
-      applicationSubmitTime: { $exists: true, $ne: null },
-    });
-    let applications = {};
-    let confirmations = {};
+    const aggregatedUsers = await ApplicationModel.aggregate([
+      {
+        $group: {
+          _id: "$status",
+          count: { $sum: 1 },
+        },
+      },
+    ]);
 
-    const applicationBranchAggregator = [
+    const aggregatedApplicationData = await ApplicationModel.aggregate([
       {
         $group: {
           _id: "$applicationBranch",
+          branchName: { $first: "$applicationBranch.name" },
+          data: { $push: "$applicationData" },
+        },
+      },
+    ]);
+
+    const aggregatedBranches = await BranchModel.aggregate([
+      {
+        $group: {
+          _id: "$name",
```

Comment:
I think when you do these aggregates you can combine multiple aggregates for each model. See [here](https://stackoverflow.com/questions/42325866/how-do-i-do-multiple-mongoose-aggregate-search) for reference.